compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1034,16 +1069,21 @@ static SizesAndScalableFlags getMatmulVectorSizes(func::FuncOp entryPointFn,
 
   // TODO: Compute vector tile sizes using heuristics.
 
-  if (isAArch64(targetAttr) && hasSMEFeature(targetAttr)) {
-    // Note: This may not pick any sizes (which will fallback to the default
-    // SVE) sizes below.
-    getMatmulAArch64SMEVectorSizes(op, matmulTileSizes, matmulScalableFlags);
+  if (isAArch64(targetAttr)) {
+    if (hasSMEFeature(targetAttr)) {
+      // Note: This may not pick any sizes (which will fallback to the default
+      // SVE) sizes below.
+      getMatmulAArch64SMEVectorSizes(op, matmulTileSizes, matmulScalableFlags);
+    } else {
+      getMatmulAArch64NeonSVEVectorSizes(entryPointFn, op, vectorSize,
```

Comment:
If getMatmulAArch64SMEVectorSizes returns empty, it should fallback to the SVE vector sizes (that's what the comment above it attempts to say). I don't think that happens here.

Reply:
Fixed code and comment